### DISCUSSION TITLE
Dynamic anvil menu

### DISCUSSION THREAD

**Discussion by mikyvars** [Odalita-Developments/OdalitaMenus#6] [Q&A] [answered]
Hello,

I'm trying to make an anvil menu, but I think I'm missing something because it's not giving me the result I wanted.
When I'm loading the menu, I set the repair cost to 0, but when I change the text, the repair cost change too, so how can I handle the text change event to keep repair costs at 0?
```
@Menu(title = "&6Renommer ton équipe", type = MenuType.ANVIL)
public final class MenuTeamRename implements PlayerMenuProvider {

    private final TeamData teamData;

    public MenuTeamRename(TeamData teamData) {
        this.teamData = teamData;
    }

    @Override
    public void onLoad(@NotNull Player player, @NotNull MenuContents menuContents) {
        menuContents.menuSession().setMenuProperty(MenuProperty.REPAIR_COST, 0);
        menuContents.set(0, DisplayItem.of(new ItemBuilder(Material.PAPER).withName(teamData.getName(false)).build()));
        menuContents.set(2, ClickableItem.of(new ItemBuilder(Material.PAPER).withName(teamData.getName(false)).build(), inventoryClickEvent -> {
            AnvilInventory anvilInventory = (AnvilInventory) inventoryClickEvent.getInventory();

            if(anvilInventory.getRenameText() != null)
                player.sendMessage(Component.text(anvilInventory.getRenameText()));
        }));
    }
}
```

BTW Thank you for this epic menu library!

**Comment by TryDev07** [comment, marked as answer]
Hello, 

We have noticed that this is an issue. We will be fixing this issue as soon as possible.
for now this isn't working but in an later update of the libary this should. I will let you know once this issue is resolved.

Pull request Reference: #8

**Reply by mikyvars** [reply]
Thank you for your answer,
Good to know, I'll wait for the fix and I'll use a command for the moment.

**Comment by kiwisap** [comment]
Hey,

The pull request has been merged, took a little while to fix it, but it is there.
I've also added an example for the anvil input menu, you can take a look at that if you want.
